build-logic/convention/src/main/kotlin/com.livefront.sealedenum.publish.gradle.kts

```diff
@@ -0,0 +1,65 @@
+plugins{
+    kotlin("jvm")
+    `maven-publish`
+    id("org.jetbrains.dokka")
+    id("org.jetbrains.kotlinx.binary-compatibility-validator")
```

Comment:
I setup dokka, validation, and publishing into the `publish` plugin since those are all inter-related for the sake of publishing.

If a module wasn't being published, then it doesn't really make sense to apply `dokka` or validate compatability.